### PR TITLE
Add timeout when watching pod events in k8s executor

### PR DESCRIPTION
**Pull request by dstandish**
If we don't set a timeout, it may hang indefinitely if there's a network issue.

### CODE REVIEW

**Review by hussein-awala — APPROVED** [review]
LGTM

File: airflow/providers/cncf/kubernetes/executors/kubernetes_executor_utils.py [hunk 1]
```diff
@@ -165,6 +163,14 @@ def _run(
 
         last_resource_version: str | None = None
 
+        # For info about k8s timeout settings see
+        # https://github.com/kubernetes-client/python/blob/94e42113a1fe5c580917decacdde879eab7406b3/examples/watch/timeout-settings.md
+        # and https://github.com/kubernetes-client/python/blob/b47caad922709350f477210317ac7f9574a72a97/kubernetes/client/api_client.py#L336-L339
```
Comment (jedcunningham) [inline airflow/providers/cncf/kubernetes/executors/kubernetes_executor_utils.py]: Why are these permanent links pointing to different versions?

File: airflow/providers/cncf/kubernetes/executors/kubernetes_executor_utils.py [hunk 2]
```diff
@@ -165,6 +163,14 @@ def _run(
...
+        request_timeout = 30
+        server_conn_timeout = 3600
+        kwargs["_request_timeout"] = request_timeout
+        kwargs["timeout_seconds"] = server_conn_timeout
```
Comment (jedcunningham) [inline airflow/providers/cncf/kubernetes/executors/kubernetes_executor_utils.py]:
```suggestion
        kwargs["_request_timeout"] = 30  # client-side timeout
        kwargs["timeout_seconds"] = 3600  # server-side timeout
```

nit

**Review by jedcunningham — APPROVED** [review]

File: airflow/providers/cncf/kubernetes/executors/kubernetes_executor_utils.py (hunk 1 above)
Comment (dstandish) [reply airflow/providers/cncf/kubernetes/executors/kubernetes_executor_utils.py]: 🤷

Comment (RNHTTR) [inline airflow/providers/cncf/kubernetes/executors/kubernetes_executor_utils.py]:
```suggestion
        # https://github.com/kubernetes-client/python/blob/v29.0.0/examples/watch/timeout-settings.md
        # and https://github.com/kubernetes-client/python/blob/v29.0.0/kubernetes/client/api_client.py#L336-L339
```

**Review by RNHTTR — APPROVED** [review]

File: airflow/providers/cncf/kubernetes/executors/kubernetes_executor_utils.py [hunk 3]
```diff
@@ -165,6 +163,14 @@ def _run(
 
         last_resource_version: str | None = None
 
+        # For info about k8s timeout settings see
+        # https://github.com/kubernetes-client/python/blob/v29.0.0/examples/watch/timeout-settings.md
+        # and https://github.com/kubernetes-client/python/blob/v29.0.0/kubernetes/client/api_client.py#L336-L339
+        request_timeout = 30
+        server_conn_timeout = 3600
+        kwargs["_request_timeout"] = request_timeout
```
Comment (dirrao) [inline airflow/providers/cncf/kubernetes/executors/kubernetes_executor_utils.py]: Why don't we use the configuration `kube_client_request_args`?

Comment (dstandish) [reply airflow/providers/cncf/kubernetes/executors/kubernetes_executor_utils.py]: this PR is about adding a default timeout for this specific call, not about letting the user optionally override the timeout; that is a separate issue / question IMHO

**Review by dirrao — APPROVED** [review]
LGTM

**Review by eladkal — APPROVED** [review]

**Comment by collinmcnulty** [comment]
Should this be marked as closes #36998 ?

**Comment by dstandish** [comment]
> Should this be marked as closes #36998 ?

it's not obvious to me how this would be connected to that issue

**Review by RNHTTR — APPROVED** [review]

File: airflow/providers/cncf/kubernetes/executors/kubernetes_executor_utils.py (hunk 2 above)
Comment (dstandish) [reply airflow/providers/cncf/kubernetes/executors/kubernetes_executor_utils.py]: i fixed the naming (request vs client)

File: airflow/providers/cncf/kubernetes/executors/kubernetes_executor_utils.py [hunk 4]
```diff
@@ -150,6 +148,14 @@ def _run(
 
         last_resource_version: str | None = None
 
+        # For info about k8s timeout settings see
+        # https://github.com/kubernetes-client/python/blob/v29.0.0/examples/watch/timeout-settings.md
+        # and https://github.com/kubernetes-client/python/blob/v29.0.0/kubernetes/client/api_client.py#L336-L339
+        client_timeout = 30
+        server_conn_timeout = 3600
+        kwargs["_request_timeout"] = client_timeout
+        kwargs["timeout_seconds"] = server_conn_timeout
```
Comment (brumi1024) [inline airflow/providers/cncf/kubernetes/executors/kubernetes_executor_utils.py:157]: These settings were configurable using the AIRFLOW__KUBERNETES_EXECUTOR__KUBE_CLIENT_REQUEST_ARGS env variable, but now they are overridden if I'm not mistaken. Is this expected?

**Pull request merged**